missilewars-plugin/src/main/java/de/butzlabben/missilewars/game/Game.java

```diff
@@ -441,14 +440,12 @@ private void checkTeamSize(Team team) {
     }
 
     public void resetGame() {
-        HandlerList.unregisterAll(listener);
-
-        stopTimer();
-
         applyForAllPlayers(this::teleportToAfterGameSpawn);
+        
+        HandlerList.unregisterAll(listener);
```

Comment:
With this change, the player also gets his player data back after the teleport, since the listeners can still handle the teleport event.

Reply:
I did not write "Attention, the order of the methods is important", because that is the case with many methods in the code. But with these new notes the importance of the order should be clear.